In [1]:
#Importing the required libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
df = pd.read_csv('data/AviationData.csv', encoding='latin1', low_memory=False)
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [3]:
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [5]:
df.isna().mean()*100

Event.Id                   0.000000
Investigation.Type         0.000000
Accident.Number            0.000000
Event.Date                 0.000000
Location                   0.058500
Country                    0.254250
Latitude                  61.320298
Longitude                 61.330423
Airport.Code              43.469946
Airport.Name              40.611324
Injury.Severity            1.124999
Aircraft.damage            3.593246
Aircraft.Category         63.677170
Registration.Number        1.481623
Make                       0.070875
Model                      0.103500
Amateur.Built              0.114750
Number.of.Engines          6.844491
Engine.Type                7.961615
FAR.Description           63.974170
Schedule                  85.845268
Purpose.of.flight          6.965991
Air.carrier               81.271023
Total.Fatal.Injuries      12.826109
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Uninjured            6.650992
Weather.Condition          5

**The columns Air.carrier and Schedule contain more than 80% null values. Need to drop them since any analysis on them will not be a representation of the dataset**

In [6]:
df.drop(columns=['Schedule','Air.carrier'], inplace=True)
df.isna().mean()*100

Event.Id                   0.000000
Investigation.Type         0.000000
Accident.Number            0.000000
Event.Date                 0.000000
Location                   0.058500
Country                    0.254250
Latitude                  61.320298
Longitude                 61.330423
Airport.Code              43.469946
Airport.Name              40.611324
Injury.Severity            1.124999
Aircraft.damage            3.593246
Aircraft.Category         63.677170
Registration.Number        1.481623
Make                       0.070875
Model                      0.103500
Amateur.Built              0.114750
Number.of.Engines          6.844491
Engine.Type                7.961615
FAR.Description           63.974170
Purpose.of.flight          6.965991
Total.Fatal.Injuries      12.826109
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Uninjured            6.650992
Weather.Condition          5.053494
Broad.phase.of.flight     30.560587
Report.Status              7

In [7]:
df['FAR.Description'].unique()

array([nan, 'Part 129: Foreign', 'Part 91: General Aviation',
       'Part 135: Air Taxi & Commuter', 'Part 125: 20+ Pax,6000+ lbs',
       'Part 121: Air Carrier', 'Part 137: Agricultural',
       'Part 133: Rotorcraft Ext. Load', 'Unknown',
       'Part 91F: Special Flt Ops.', 'Non-U.S., Non-Commercial',
       'Public Aircraft', 'Non-U.S., Commercial', 'Public Use',
       'Armed Forces', 'Part 91 Subpart K: Fractional', '091', 'NUSC',
       '135', 'NUSN', '121', '137', '129', '133', '091K', 'UNK', 'PUBU',
       'ARMF', '103', '125', '437', '107'], dtype=object)

In [8]:
df['Purpose.of.flight'].unique()

array(['Personal', nan, 'Business', 'Instructional', 'Unknown', 'Ferry',
       'Executive/corporate', 'Aerial Observation', 'Aerial Application',
       'Public Aircraft', 'Skydiving', 'Other Work Use', 'Positioning',
       'Flight Test', 'Air Race/show', 'Air Drop',
       'Public Aircraft - Federal', 'Glider Tow',
       'Public Aircraft - Local', 'External Load',
       'Public Aircraft - State', 'Banner Tow', 'Firefighting',
       'Air Race show', 'PUBS', 'ASHO', 'PUBL'], dtype=object)

**FAR.Description column contains 63% null values. 
Even though it provides the type of flight operation based on Federal Aviation Regulations (FARs)
which are applied in classifying how aircrafts are used be it commercial, military, agricultural.
This information is can be obtained from the purpose of flight. Furthermore, working with the column FAR.Description will need
use to do mapping of the codes i.e. 137 represents agricultural**

In [9]:
#dropping the column 'FAR.Description'
df.drop(columns = ['FAR.Description'], inplace = True)
df.isna().mean()*100

Event.Id                   0.000000
Investigation.Type         0.000000
Accident.Number            0.000000
Event.Date                 0.000000
Location                   0.058500
Country                    0.254250
Latitude                  61.320298
Longitude                 61.330423
Airport.Code              43.469946
Airport.Name              40.611324
Injury.Severity            1.124999
Aircraft.damage            3.593246
Aircraft.Category         63.677170
Registration.Number        1.481623
Make                       0.070875
Model                      0.103500
Amateur.Built              0.114750
Number.of.Engines          6.844491
Engine.Type                7.961615
Purpose.of.flight          6.965991
Total.Fatal.Injuries      12.826109
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Uninjured            6.650992
Weather.Condition          5.053494
Broad.phase.of.flight     30.560587
Report.Status              7.178616
Publication.Date          15

In [10]:
df['Aircraft.Category'].unique()

array([nan, 'Airplane', 'Helicopter', 'Glider', 'Balloon', 'Gyrocraft',
       'Ultralight', 'Unknown', 'Blimp', 'Powered-Lift', 'Weight-Shift',
       'Powered Parachute', 'Rocket', 'WSFT', 'UNK', 'ULTR'], dtype=object)

In [11]:
df['Aircraft.Category'].value_counts()

Airplane             27617
Helicopter            3440
Glider                 508
Balloon                231
Gyrocraft              173
Weight-Shift           161
Powered Parachute       91
Ultralight              30
Unknown                 14
WSFT                     9
Powered-Lift             5
Blimp                    4
UNK                      2
ULTR                     1
Rocket                   1
Name: Aircraft.Category, dtype: int64

In [12]:
for col in df.columns:
  print(f"Unique values in column '{col}':")
  print(df[col].unique())
  print('\n********')

Unique values in column 'Event.Id':
['20001218X45444' '20001218X45447' '20061025X01555' ... '20221227106497'
 '20221227106498' '20221230106513']

********
Unique values in column 'Investigation.Type':
['Accident' 'Incident']

********
Unique values in column 'Accident.Number':
['SEA87LA080' 'LAX94LA336' 'NYC07LA005' ... 'WPR23LA075' 'WPR23LA076'
 'ERA23LA097']

********
Unique values in column 'Event.Date':
['1948-10-24' '1962-07-19' '1974-08-30' ... '2022-12-22' '2022-12-26'
 '2022-12-29']

********
Unique values in column 'Location':
['MOOSE CREEK, ID' 'BRIDGEPORT, CA' 'Saltville, VA' ... 'San Manual, AZ'
 'Auburn Hills, MI' 'Brasnorte, ']

********
Unique values in column 'Country':
['United States' nan 'GULF OF MEXICO' 'Puerto Rico' 'ATLANTIC OCEAN'
 'HIGH ISLAND' 'Bahamas' 'MISSING' 'Pakistan' 'Angola' 'Germany'
 'Korea, Republic Of' 'Martinique' 'American Samoa' 'PACIFIC OCEAN'
 'Canada' 'Bolivia' 'Mexico' 'Dominica' 'Netherlands Antilles' 'Iceland'
 'Greece' 'Guam' 'Australia' '

In [13]:
df['Event.Date'].value_counts()

1982-05-16    25
1984-06-30    25
2000-07-08    25
1983-08-05    24
1983-06-05    24
              ..
1990-12-23     1
2005-11-29     1
2010-11-16     1
2020-04-17     1
1996-10-23     1
Name: Event.Date, Length: 14782, dtype: int64

In [14]:
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type',
       'Purpose.of.flight', 'Total.Fatal.Injuries', 'Total.Serious.Injuries',
       'Total.Minor.Injuries', 'Total.Uninjured', 'Weather.Condition',
       'Broad.phase.of.flight', 'Report.Status', 'Publication.Date'],
      dtype='object')

In [15]:
df['Event.Date'].dtype

dtype('O')

In [16]:
#Coverting the event.date into a datetime type
df['Event.Date'] = pd.to_datetime(df['Event.Date'], errors='coerce')

In [17]:
df['Publication.Date'].unique()

array([nan, '19-09-1996', '26-02-2007', ..., '22-12-2022', '23-12-2022',
       '29-12-2022'], dtype=object)

In [18]:
#The column Publication.Date contains null values, we can inpute using a date like 01-01-1901 and then we convert to datetime
df['Publication.Date'].fillna('01-01-1901', inplace=True)

In [19]:
df['Publication.Date'].unique()

array(['01-01-1901', '19-09-1996', '26-02-2007', ..., '22-12-2022',
       '23-12-2022', '29-12-2022'], dtype=object)

In [20]:
df['Publication.Date'].dtype

dtype('O')

In [21]:
df['Publication.Date'] = pd.to_datetime(df['Publication.Date'], errors='coerce')
df['Publication.Date'].dtype

dtype('<M8[ns]')

In [22]:
df['Location'].isna().sum()

52

In [23]:
df['Country'].unique()

array(['United States', nan, 'GULF OF MEXICO', 'Puerto Rico',
       'ATLANTIC OCEAN', 'HIGH ISLAND', 'Bahamas', 'MISSING', 'Pakistan',
       'Angola', 'Germany', 'Korea, Republic Of', 'Martinique',
       'American Samoa', 'PACIFIC OCEAN', 'Canada', 'Bolivia', 'Mexico',
       'Dominica', 'Netherlands Antilles', 'Iceland', 'Greece', 'Guam',
       'Australia', 'CARIBBEAN SEA', 'West Indies', 'Japan',
       'Philippines', 'Venezuela', 'Bermuda', 'San Juan Islands',
       'Colombia', 'El Salvador', 'United Kingdom',
       'British Virgin Islands', 'Netherlands', 'Costa Rica',
       'Mozambique', 'Jamaica', 'Panama', 'Guyana', 'Norway', 'Hong Kong',
       'Portugal', 'Malaysia', 'Turks And Caicos Islands',
       'Northern Mariana Islands', 'Dominican Republic', 'Suriname',
       'Honduras', 'Congo', 'Belize', 'Guatemala', 'Anguilla', 'France',
       'St Vincent And The Grenadines', 'Haiti', 'Montserrat',
       'Papua New Guinea', 'Cayman Islands', 'Sweden', 'Taiwan',
       'Se

In [24]:
#going through the list above, the names below are not indicative of any country in particular
invalid_names = [ 'GULF OF MEXICO', 'ATLANTIC OCEAN', 'HIGH ISLAND', 'PACIFIC OCEAN',
    'CARIBBEAN SEA', 'San Juan Islands', 'BLOCK 651A', 'Central Africa',
    'Wolseley', 'Obyan', 'AY', 'Cayenne', 'UN']

df['Country'] = df['Country'].replace(invalid_names, 'Unknown')

country_name_var = {
    'Ivory Coast': "Côte d'Ivoire",
    "Cote D'ivoire": "Côte d'Ivoire",
    'South Korean': 'South Korea',
    'Korea, Republic Of': 'South Korea',
    'Bosnia And Herzegovina': 'Bosnia and Herzegovina',
    'St Vincent And The Grenadines': 'Saint Vincent and the Grenadines',
    'Antigua And Barbuda': 'Antigua and Barbuda',
    'Turks And Caicos': 'Turks and Caicos Islands',
    'Turks And Caicos Islands': 'Turks and Caicos Islands',
    'Great Britain': 'United Kingdom',
    'Scotland': 'United Kingdom',
}

df['Country'] = df['Country'].replace(country_name_var)
df['Country'] = df['Country'].fillna('Unknown')


In [25]:
df['Country'].isna().sum()

0

In [26]:
df['Airport.Code'].isna().mean()*100

43.46994566256792

In [27]:
df['Airport.Code'].fillna('UNK', inplace=True)

In [28]:
df['Airport.Code'].isna().mean()*100

0.0

In [29]:
df['Airport.Name'].isna().mean()*100

40.611324235844705

In [30]:
df['Airport.Name'].unique()

array([nan, 'BLACKBURN AG STRIP', 'HANOVER', ...,
       'HAWKINSVILLE-PULASKI COUNTY', 'Lewiston Municipal Airport',
       'WICHITA DWIGHT D EISENHOWER NT'], dtype=object)

In [31]:
df['Airport.Name'].fillna('UNKNOWN', inplace=True)
df['Airport.Name'].isna().mean()*100

0.0

In [32]:
df['Weather.Condition'].unique()

array(['UNK', 'IMC', 'VMC', nan, 'Unk'], dtype=object)

**Need to format the UNK and Unk into one and replace the missing values with UNK string since the column is using abbreviation**

In [33]:
df['Weather.Condition'] = df['Weather.Condition'].replace({'Unk': 'UNK'})
df['Weather.Condition'].fillna('UNK', inplace=True)
df['Weather.Condition'].value_counts()

VMC    77303
IMC     5976
UNK     5610
Name: Weather.Condition, dtype: int64

In [34]:
df['Report.Status'].isna().mean()*100

7.1786160267299675

**The report status only contains 7% null values, We can drop this null rows***

In [35]:
df = df.dropna(subset=['Report.Status'])
df['Report.Status'].isna().mean()*100

0.0

In [36]:
## Current status of the dataframe info

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 82508 entries, 0 to 88767
Data columns (total 28 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Event.Id                82508 non-null  object        
 1   Investigation.Type      82508 non-null  object        
 2   Accident.Number         82508 non-null  object        
 3   Event.Date              82508 non-null  datetime64[ns]
 4   Location                82456 non-null  object        
 5   Country                 82508 non-null  object        
 6   Latitude                30388 non-null  object        
 7   Longitude               30378 non-null  object        
 8   Airport.Code            82508 non-null  object        
 9   Airport.Name            82508 non-null  object        
 10  Injury.Severity         82417 non-null  object        
 11  Aircraft.damage         80359 non-null  object        
 12  Aircraft.Category       26328 non-null  object

In [38]:
df['Injury.Severity'].unique()

array(['Fatal(2)', 'Fatal(4)', 'Fatal(3)', 'Fatal(1)', 'Non-Fatal',
       'Incident', 'Fatal(8)', 'Fatal(78)', 'Fatal(7)', 'Fatal(6)',
       'Fatal(5)', 'Fatal(153)', 'Fatal(12)', 'Fatal(14)', 'Fatal(23)',
       'Fatal(10)', 'Fatal(11)', 'Fatal(9)', 'Fatal(17)', 'Fatal(13)',
       'Fatal(29)', 'Fatal(70)', 'Unavailable', 'Fatal(135)', 'Fatal(31)',
       'Fatal(256)', 'Fatal(25)', 'Fatal(82)', 'Fatal(156)', 'Fatal(28)',
       'Fatal(18)', 'Fatal(43)', 'Fatal(15)', 'Fatal(270)', 'Fatal(144)',
       'Fatal(174)', 'Fatal(111)', 'Fatal(131)', 'Fatal(20)', 'Fatal(73)',
       'Fatal(27)', 'Fatal(34)', 'Fatal(87)', 'Fatal(30)', 'Fatal(16)',
       'Fatal(47)', 'Fatal(56)', 'Fatal(37)', 'Fatal(132)', 'Fatal(68)',
       'Fatal(54)', 'Fatal(52)', 'Fatal(65)', 'Fatal(72)', 'Fatal(160)',
       'Fatal(189)', 'Fatal(123)', 'Fatal(33)', 'Fatal(110)',
       'Fatal(230)', 'Fatal(97)', 'Fatal(349)', 'Fatal(125)', 'Fatal(35)',
       'Fatal(228)', 'Fatal(75)', 'Fatal(104)', 'Fatal(229)', 'Fatal

In [39]:
df['Injury.Severity'].isna().mean()*100

0.11029233528869929

**The Injury.Severity column only contains 0.11% null values, We can drop these null rows**

In [40]:
df = df.dropna(subset=['Injury.Severity'])
df['Injury.Severity'].isna().mean()*100

0.0

In [41]:
print(df['Injury.Severity'].unique())

['Fatal(2)' 'Fatal(4)' 'Fatal(3)' 'Fatal(1)' 'Non-Fatal' 'Incident'
 'Fatal(8)' 'Fatal(78)' 'Fatal(7)' 'Fatal(6)' 'Fatal(5)' 'Fatal(153)'
 'Fatal(12)' 'Fatal(14)' 'Fatal(23)' 'Fatal(10)' 'Fatal(11)' 'Fatal(9)'
 'Fatal(17)' 'Fatal(13)' 'Fatal(29)' 'Fatal(70)' 'Unavailable'
 'Fatal(135)' 'Fatal(31)' 'Fatal(256)' 'Fatal(25)' 'Fatal(82)'
 'Fatal(156)' 'Fatal(28)' 'Fatal(18)' 'Fatal(43)' 'Fatal(15)' 'Fatal(270)'
 'Fatal(144)' 'Fatal(174)' 'Fatal(111)' 'Fatal(131)' 'Fatal(20)'
 'Fatal(73)' 'Fatal(27)' 'Fatal(34)' 'Fatal(87)' 'Fatal(30)' 'Fatal(16)'
 'Fatal(47)' 'Fatal(56)' 'Fatal(37)' 'Fatal(132)' 'Fatal(68)' 'Fatal(54)'
 'Fatal(52)' 'Fatal(65)' 'Fatal(72)' 'Fatal(160)' 'Fatal(189)'
 'Fatal(123)' 'Fatal(33)' 'Fatal(110)' 'Fatal(230)' 'Fatal(97)'
 'Fatal(349)' 'Fatal(125)' 'Fatal(35)' 'Fatal(228)' 'Fatal(75)'
 'Fatal(104)' 'Fatal(229)' 'Fatal(80)' 'Fatal(217)' 'Fatal(169)'
 'Fatal(88)' 'Fatal(19)' 'Fatal(60)' 'Fatal(113)' 'Fatal(143)' 'Fatal(83)'
 'Fatal(24)' 'Fatal(44)' 'Fatal(64)' 'Fatal(92

In [42]:
#let us create another column with the severity class
def Severity_class(injury_severity):
    if 'Fatal' in injury_severity:
        return 'Fatal'
    elif injury_severity in ['Non-Fatal', 'Minor']:
        return 'Non-Fatal'
    elif injury_severity in ['Serious']:
        return 'Serious'
    elif injury_severity == 'Incident':
        return 'Incident'
    else:
        return 'Unknown'
df['Severity_class'] = df['Injury.Severity'].apply(Severity_class)

In [43]:
df.shape

(82417, 29)

In [44]:
df['Aircraft.damage'].unique()

array(['Destroyed', 'Substantial', 'Minor', nan, 'Unknown'], dtype=object)

In [45]:
df['Aircraft.damage'].isna().mean()*100

2.5504446898091415

**The Aircraft.damage column only contains 2.55% null values, We can impute with Unknown string**

In [46]:
df['Aircraft.damage'].fillna('Unknown', inplace=True)
df['Aircraft.damage'].value_counts()

Substantial    60708
Destroyed      17287
Minor           2313
Unknown         2109
Name: Aircraft.damage, dtype: int64

In [47]:
df['Aircraft.Category'].unique()

array([nan, 'Airplane', 'Helicopter', 'Glider', 'Balloon', 'Gyrocraft',
       'Ultralight', 'Unknown', 'Blimp', 'Powered-Lift', 'Weight-Shift',
       'Powered Parachute', 'Rocket', 'WSFT'], dtype=object)

In [48]:
df['Aircraft.Category'].value_counts()

Airplane             22563
Helicopter            2577
Glider                 459
Balloon                216
Weight-Shift           151
Gyrocraft              150
Powered Parachute       87
Ultralight              25
Unknown                  5
Powered-Lift             4
Blimp                    4
WSFT                     1
Rocket                   1
Name: Aircraft.Category, dtype: int64

**The 'Aircraft.Category' column contains nan values and a preexisting Unknown String, we can impute using the preexisting string**

In [49]:
df['Aircraft.Category'].fillna('Unknown', inplace=True)
df['Aircraft.Category'].value_counts()

Unknown              56179
Airplane             22563
Helicopter            2577
Glider                 459
Balloon                216
Weight-Shift           151
Gyrocraft              150
Powered Parachute       87
Ultralight              25
Powered-Lift             4
Blimp                    4
WSFT                     1
Rocket                   1
Name: Aircraft.Category, dtype: int64

In [50]:
df['Registration.Number'].isna().mean()*100

1.5518642998410521

**The 'Registration.Number' column only contains 1.55% null values, We can drop these null rows**

In [51]:
df = df.dropna(subset=['Registration.Number'])

In [52]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 81138 entries, 0 to 88767
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Event.Id                81138 non-null  object        
 1   Investigation.Type      81138 non-null  object        
 2   Accident.Number         81138 non-null  object        
 3   Event.Date              81138 non-null  datetime64[ns]
 4   Location                81108 non-null  object        
 5   Country                 81138 non-null  object        
 6   Latitude                29958 non-null  object        
 7   Longitude               29948 non-null  object        
 8   Airport.Code            81138 non-null  object        
 9   Airport.Name            81138 non-null  object        
 10  Injury.Severity         81138 non-null  object        
 11  Aircraft.damage         81138 non-null  object        
 12  Aircraft.Category       81138 non-null  object

**The columns ['Make','Model','Amateur.Built'] contain few null rows: Decision: Drop the null rows**

In [54]:
df = df.dropna(subset=['Make','Model','Amateur.Built' ])

In [56]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 81060 entries, 0 to 88767
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Event.Id                81060 non-null  object        
 1   Investigation.Type      81060 non-null  object        
 2   Accident.Number         81060 non-null  object        
 3   Event.Date              81060 non-null  datetime64[ns]
 4   Location                81031 non-null  object        
 5   Country                 81060 non-null  object        
 6   Latitude                29929 non-null  object        
 7   Longitude               29919 non-null  object        
 8   Airport.Code            81060 non-null  object        
 9   Airport.Name            81060 non-null  object        
 10  Injury.Severity         81060 non-null  object        
 11  Aircraft.damage         81060 non-null  object        
 12  Aircraft.Category       81060 non-null  object

**Start from column 18 to 25 replacing the categorical variables then later do the numerical**